cos-fleetshard-operator/src/main/java/org/bf2/cos/fleetshard/operator/connector/ConnectorController.java

```diff
@@ -393,6 +403,50 @@ private UpdateControl<ManagedConnector> handleAugmentation(ManagedConnector conn
             .withName(connector.getSpec().getDeployment().getSecret())
             .get();
 
+        ConfigMap configMap = kubernetesClient.configMaps()
+            .inNamespace(connector.getMetadata().getNamespace())
+            .withName(ConfigMaps.generateConnectorConfigMapId(connector.getSpec().getDeploymentId()))
+            .get();
+
+        if (configMap == null) {
+            LOGGER.info(
+                "Configmap not found (cluster_id: {}, namespace_id: {}, connector_id: {}, deployment_id: {}), creating a new one",
+                connector.getSpec().getClusterId(),
+                connector.getMetadata().getNamespace(),
+                connector.getSpec().getConnectorId(),
+                connector.getSpec().getDeploymentId());
+
+            configMap = new ConfigMap();
+            configMap.setMetadata(new ObjectMeta());
+            configMap.getMetadata().setNamespace(connector.getMetadata().getNamespace());
+            configMap.getMetadata().setName(ConfigMaps.generateConnectorConfigMapId(connector.getSpec().getDeploymentId()));
+
+            Resources.setLabels(
+                configMap,
+                LABEL_CLUSTER_ID, connector.getSpec().getClusterId(),
+                LABEL_CONNECTOR_ID, connector.getSpec().getConnectorId(),
+                LABEL_DEPLOYMENT_ID, connector.getSpec().getDeploymentId());
+
+            Resources.setOwnerReferences(
+                configMap,
+                connector);
+
+            // copy operator type
+            Resources.setLabel(
+                secret,
+                LABEL_OPERATOR_TYPE,
+                connector.getMetadata().getLabels().get(LABEL_OPERATOR_TYPE));
+            try {
+                this.kubernetesClient.configMaps()
+                    .inNamespace(configMap.getMetadata().getNamespace())
+                    .withName(configMap.getMetadata().getName())
+                    .createOrReplace(configMap);
```

Comment:
This should probably be just `create` instead of `createOrReplace` as it may override an existing configmap in case it gets created between the check for presence and the invocation of `createOrReplace`

Reply:
done